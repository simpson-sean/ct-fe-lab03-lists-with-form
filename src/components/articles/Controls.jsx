import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ topic, onTopicSearch, onSubmit }) => (
    <form onSubmit={onSubmit} data-testid="news-form">
        <label htmlFor="topic">Topic Search</label>
        <input
            id="topic"
            type="text"
            value={topic}
            onChange={onTopicSearch}
        />
        <button aria-label='find-articles'>Search Articles</button>
    </form>
);

Controls.propTypes = {
    topic: PropTypes.string.isRequired,
    onTopicSearch: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired, 
};

export default Controls;