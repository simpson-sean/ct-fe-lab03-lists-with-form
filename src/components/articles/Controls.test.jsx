import React from 'react';
import { render, screen } from '@testing-library/react';
import Controls from './Controls';

describe('Controls', () => {
    it('displays a form with an input and submit button', () => {
      render(
        <Controls
          topic="JavaScript"
          onTopicSearch={() => {}}
          onSubmit={() => {}}
        />
      );
  
      const controlsForm = screen.getByTestId('news-form');
      expect(controlsForm).toMatchSnapshot();
    });
  });