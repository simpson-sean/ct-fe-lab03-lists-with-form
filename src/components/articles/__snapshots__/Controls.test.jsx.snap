// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls displays a form with an input and submit button 1`] = `
<form
  data-testid="news-form"
>
  <label
    for="topic"
  >
    Topic Search
  </label>
  <input
    id="topic"
    type="text"
    value="JavaScript"
  />
  <button
    aria-label="find-articles"
  >
    Search Articles
  </button>
</form>
`;
